packages/builders/src/components/selectMenu/UnsafeSelectMenu.ts

```diff
@@ -1,45 +1,60 @@
 import { ComponentType, type APISelectMenuComponent } from 'discord-api-types/v9';
-import type { Component } from '../Component';
-import { SelectMenuOption } from './SelectMenuOption';
+import { Component } from '../Component';
+import { UnsafeSelectMenuOption } from './UnsafeSelectMenuOption';
 
 /**
  * Represents a non-validated select menu component
  */
-export class UnsafeSelectMenuComponent implements Component {
-	public readonly type = ComponentType.SelectMenu as const;
-	public readonly options: SelectMenuOption[];
-	public readonly placeholder?: string;
-	public readonly min_values?: number;
-	public readonly max_values?: number;
-	public readonly custom_id!: string;
-	public readonly disabled?: boolean;
+export class UnsafeSelectMenuComponent extends Component {
+	protected declare data: APISelectMenuComponent;
+	public readonly options: UnsafeSelectMenuOption[] = [];
 
 	public constructor(data?: APISelectMenuComponent) {
-		this.options = data?.options.map((option) => new SelectMenuOption(option)) ?? [];
-		this.placeholder = data?.placeholder;
-		this.min_values = data?.min_values;
-		this.max_values = data?.max_values;
-		/* eslint-disable @typescript-eslint/non-nullable-type-assertion-style */
-		this.custom_id = data?.custom_id as string;
-		/* eslint-enable @typescript-eslint/non-nullable-type-assertion-style */
-		this.disabled = data?.disabled;
+		super(data);
+		this.data.type ??= ComponentType.SelectMenu;
+		this.data.options ??= [];
```

Comment:
Please do not mutate input, and since we're mutating it... let's clone it.
```suggestion
		super({ type: ComponentType.SelectMenu, optioms: [], ...data });
```